plugins/permCheck.ts

```diff
@@ -19,21 +26,296 @@
  * @end
  */
 
-import type { GuildMember, PermissionResolvable } from "discord.js";
-import { CommandControlPlugin, CommandType, controller } from "@sern/handler";
-export function permCheck(perm: PermissionResolvable, response: string) {
-	return CommandControlPlugin<CommandType.Both>(async (ctx, args) => {
+import {
+	ChatInputCommandInteraction,
+	Colors,
+	EmbedBuilder,
+	PermissionsBitField,
+	type GuildMember,
+	type PermissionResolvable,
+	type TextChannel,
+} from "discord.js";
+import {
+	CommandControlPlugin,
+	CommandType,
+	controller,
+	Service,
+} from "@sern/handler";
+
+function command(perm: PermissionResolvable, response?: string) {
+	return CommandControlPlugin<CommandType.Both>(async (ctx) => {
 		if (ctx.guild === null) {
-			await ctx.reply("This command cannot be used here");
-			console.warn(
-				"PermCheck > A command stopped because we couldn't check a users permissions (was used in dms)",
-			); //delete this line if you dont want to be notified when a command is used outside of a guild/server
+			await ctx.reply("This command cannot be used in DM's!");
+			return controller.stop();
+		}
+		if (
+			!(ctx.member! as GuildMember)
+				.permissionsIn(ctx.channel as TextChannel)
+				.has(perm)
+		) {
+			await ctx.reply({
+				embeds: [
+					sendEmbed(
+						response ??
+							`You are missing required permissions to run this command:\n${permsToString(
+								perm
+							)}`
+					),
+				],
+				ephemeral: !ctx.isMessage() ? true : false,
+			});
 			return controller.stop();
 		}
-		if (!(ctx.member! as GuildMember).permissions.has(perm)) {
-			await ctx.reply(response);
+		if (
+			!(ctx.member! as GuildMember)
+				.permissionsIn(ctx.channel as TextChannel)
+				.any(perm)
+		) {
+			await ctx.reply({
+				embeds: [
+					sendEmbed(
+						response ??
+							`You need at least one of the following permissions to run this command:\n${permsToString(
+								perm
+							)}`
+					),
+				],
+				ephemeral: ctx.isMessage() ? false : true,
+			});
 			return controller.stop();
 		}
 		return controller.next();
 	});
 }
+function subGroups(opts: BaseOptions[]) {
+	return CommandControlPlugin<CommandType.Slash>(async ({ interaction }) => {
```

Comment:
destructuring the interaction here will make this plugin probably fail for Both Commands that run with a text command.